### PR TITLE
[12.0][ADD] sale_timesheet_role: set product and rate according to employee/role

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Adds support for `hr_timesheet_role` by introducing feature of specifying product per employee/role combination. Depends on #762
image
image

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@pedrobaeza we've discussed this module a bit, so you're most aware of the topic
@aheficent I guess you have interest in this module as well

**Comment by pedrobaeza** [comment]
Please expand more the README for being aware of all the details, as it's now very brief.

**Comment by alexey-pelykh** [comment]
@pedrobaeza you mean the module readme or the PR description?

**Comment by pedrobaeza** [comment]
The module README, although when being a new module, I always put the main comment with the same README text as a good habit.

**Comment by rafaelbn** [comment]
Hi @alexey-pelykh , it important to test everything independently module by module. Here you are adding  sale_timesheet_enhanced in order to test this module but usually is better to wait until the dependencies are merged. For me "the tester" is confusing.

File: sale_timesheet_enhanced/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    'website': 'https://github.com/OCA/sale-workflow',
+    'author':
+        'Brainbean Apps, '
+        'Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'installable': True,
+    'application': False,
+    'summary': 'Enhanced selling timesheets in your sales order',
+    'depends': [
+        'sale_timesheet',
```
Comment (rafaelbn) [inline sale_timesheet_enhanced/__manifest__.py:17]: Here is missing the dependency to https://github.com/OCA/sale-workflow/pull/762

**Comment by rafaelbn** [comment]
Hi @alexey-pelykh This PR includes sale_timesheet_enhanced and it musn't . sale_timesheet_role depends in sale_timesheet_enhaced as you told.

I will try to test if I get some time, thank you!

File: sale_timesheet_enhanced/__manifest__.py (hunk 1 above)
Comment (rafaelbn) [reply sale_timesheet_enhanced/__manifest__.py:17]: Here is missing the dependency to sale_timesheet_enhanced

**Comment by alexey-pelykh** [comment]
@rafaelbn that's temporary until `sale_timesheet_enhaced` is reviewed and merged, then this PR will be rebased

**Comment by alexey-pelykh** [comment]
It's impossible to implement this module w/o improvements introduced by `sale_timesheet_enhanced`

**Comment by alexey-pelykh** [comment]
Superseded by https://github.com/OCA/timesheet/pull/248

**Pull request closed without merging**